app/controllers/supplementing_content_controller.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class SupplementingContentController < ApplicationController
+  def show
+    file_set_document = SolrDocument.find(params[:id])
+    digest = file_set_document['digest_ssim'].first
+
+    return render plain: 'No content found', status: :not_found if digest.blank?
+
+    s3_object = Aws::S3::Object.new(aws_bucket, digest.gsub('urn:sha1:', ''))
+
+    return render plain: 'File not found', status: :not_found unless s3_object.exists?
+
+    begin
+      content = s3_object.get.body.read
```

Comment:
this loads the whole file in to memory. if the file is 6GB it will take 6GB of memory. If the file is 200GB it will just crash over and over again.

Reply:
sounds good, i'll look into, just FYI vtt files (which is what i'm dealing with) are probably gonna be around the 100kb range we we're talking pretty small, I was also thinking about indexing the content onto the fileset itself since we index OCR text anyways as well